internal/http/healthcheck_server_test.go

```diff
@@ -0,0 +1,166 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestHealthCheckHandler(t *testing.T) {
+	tests := []struct {
+		name           string
+		method         string
+		expectedStatus int
+		wantBody       bool
+	}{
+		{
+			name:           "GET request returns 200 and status ok",
+			method:         http.MethodGet,
+			expectedStatus: http.StatusOK,
+			wantBody:       true,
+		},
+		{
+			name:           "POST request returns 405 and status not allowed",
+			method:         http.MethodPost,
+			expectedStatus: http.StatusMethodNotAllowed,
+			wantBody:       false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req := httptest.NewRequest(tt.method, "/healthz", nil)
+			w := httptest.NewRecorder()
+
+			handleHealthCheck(w, req)
+
+			if got := w.Code; got != tt.expectedStatus {
+				t.Errorf("handleHealthCheck() status = %v, want %v", got, tt.expectedStatus)
+			}
```

Comment:
This comes a bit late to the party since most of the tests have already been written, but maybe we should use a package like [`github.com/stretchr/testify/assert`](https://github.com/stretchr/testify) to make these assertions clearer? E.g. this would then be

```suggestion
			assert.Equal(t, w.Code, tt.expectedStatus, "wrong handleHealthCheck() status")
```

Reply:
Good idea, I'll start using `testify` from now on and maybe later we can revisit older tests!